SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -50,14 +62,21 @@ bool ComponentFactory::FillSchemaObject(Schema_Object* ComponentObject, UObject*
 			const FRepParentCmd& Parent = Changes.RepLayout.Parents[Cmd.ParentIndex];
 
 #if TRACE_LIB_ACTIVE
-			if (LatencyTracer != nullptr)
+			if (LatencyTracer != nullptr && OutLatencyTraceId != nullptr)
 			{
-				OutLatencyTraceId = LatencyTracer->RetrievePendingTrace(Object, Cmd.Property);
-				if (OutLatencyTraceId == USpatialLatencyTracer::InvalidTraceKey)
+				TraceKey PropertyKey = USpatialLatencyTracer::InvalidTraceKey;
+				PropertyKey = LatencyTracer->RetrievePendingTrace(Object, Cmd.Property);
+				if (PropertyKey == USpatialLatencyTracer::InvalidTraceKey)
 				{
 					// Check for sending a nested property
-					OutLatencyTraceId = LatencyTracer->RetrievePendingTrace(Object, Parent.Property);
+					PropertyKey = LatencyTracer->RetrievePendingTrace(Object, Parent.Property);
 				}
+				// If we have already got a trace for this actor/component, we will end one of them here
+				if (*OutLatencyTraceId != USpatialLatencyTracer::InvalidTraceKey)
+				{
+					LatencyTracer->EndLatencyTrace(*OutLatencyTraceId, TEXT("Multiple actor component traces not supported"));
+				}
+				*OutLatencyTraceId = PropertyKey;
```

Comment:
Wait, won't this override any trace that was active with an invalid trace key?

Reply:
+1